persist/fs/seek.go

```diff
@@ -181,6 +181,13 @@ func (s *seeker) Open(namespace ts.ID, shard uint32, blockStart time.Time) error
 		return err
 	}
 
+	if !s.keepUnreadBuf {
```

Comment:
Why move it inside `Open`? The block retrieve manager explicitly reuses the buffer across multiple `Open` calls to reduce memory allocation and GC, and moving it inside `Open` defeats this purpose